.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,101 @@
+                    - raspberrypi0-wifi
+                    - raspberrypi3-64
+                    - raspberrypi4-64
+                    - verdin-am62
+                    - verdin-imx8mm
+                    - verdin-imx8mp
+            branch:
+                description: "Manifest repo branch to use"
+                required: true
+                type: choice
+                options:
+                    - kirkstone
+                    - master
+                    - nanbield
+            manifest:
+                description: "Manifest file to use"
+                required: true
+                type: choice
+                options:
+                    - common.xml
+                    - default.xml
+                    - integration.xml
+                    - next.xml
+
+env:
+  BUCKET_NAME: "s3://commontorizon/ostree-repo/"
+  AWS_REGION: "eu-central-1"
+  MACHINE: "${{ inputs.machine }}"
+  BRANCH: "${{ inputs.branch }}"
+  MANIFEST: "${{ inputs.manifest }}"
+
+permissions:
+    id-token: write   # This is required for requesting the JWT
+    contents: read    # This is required for actions/checkout
+
+jobs:
+  build-yocto:
+    runs-on: [self-hosted, linux, x64, yocto]
+    steps:
+    - name: Build Yocto image
+      working-directory: /workdir
+      run:  |
+        repo init -u https://github.com/commontorizon/commontorizon-manifest.git -b ${{ env.BRANCH }} -m torizoncore/${{ env.MANIFEST }}
+        repo sync
+        MACHINE=${{ env.MACHINE }} EULA=1 source setup-environment 
+        bitbake torizon-core-docker
+    - name: Create tar archive of artifacts
+      run: |
+        tar -cvf ${{ env.MACHINE }}-artifacts.tar -C /workdir/build-*/deploy/images/${{ env.MACHINE }}/ .
+    - name: Upload Yocto artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ env.MACHINE }}-artifacts
+        path: ${{ env.MACHINE }}-artifacts.tar
+
+  ostree-push:
+    runs-on: [self-hosted, linux, x64, yocto]
+    needs: build-yocto
+    steps:
+    - name: Download Yocto artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: ${{ env.MACHINE }}-artifacts
+    - name: Unpack artifacts
+      run: |
+        tar xf ${{ env.MACHINE }}-artifacts.tar
+    - name: Configure aws credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: arn:aws:iam::590183833123:role/CommonTorizonOstreePushToS3
+        role-session-name: CommonTorizonOSTreePushSession
+        aws-region: ${{ env.AWS_REGION }}
+    - name: Sync OSTree to S3
+      run: |
+        aws s3 sync ostree_repo ${{ env.BUCKET_NAME }} --exclude='*' --include='objects/*' --include='deltas/*' --size-only
+        aws s3 sync ostree_repo ${{ env.BUCKET_NAME }} --exclude='*' --include='refs/*' --include='summar*' --include 'config*' --delete
+        aws s3 sync ostree_repo ${{ env.BUCKET_NAME }} --exclude='*' --include='objects/*' --include='deltas/*' --size-only --delete
```

Comment:
As written, it looks like this will overwrite the repo in the s3 bucket with each new build you do, so there will only ever be one version in the repo.

These s3 sync commands you copied from the matrix build aren't syncing the ostree repo that bitbake builds. It syncs from the persistent repo that toradex maintains--i.e. the repo after all promotions/`ostree pull-local`s are done. It's an unfortunate limitation of ostree that there's no push protocol--only pull. So you need to do the yocto build, pull your commit from the resultant repo into your persistent repo, then sync it to s3. I've occasionally been tempted to try out mounting the s3 bucket that the repo is ultimately hosted in using some kind of fuse drive (last I looked there were a couple options), and then being able to do a "pull" directly from the yocto-built repo to the s3 repo--maybe you could look into that?

Reply:
I'm already aware of this and am looking into this on the side. I opened this PR to start gathering feedback on implementation details since I didn't really have any concrete requirements from the CommonTorizon side on how they want the pipeline to work. There's also many details to discuss like runner setup and such before this workflow can even be used in this repo, which is why I opened this sooner rather than later.

Reply:
@jsrc27 the machine to run the self-hosted runner is configured. The work here LGTM. The only point missing is about how to handle this issue pointed by the @tkfu